platform/sync-installer/sync-installer-impl/src/main/java/org/codice/ddf/sync/installer/impl/SynchronisedInstallerImpl.java

```diff
@@ -0,0 +1,566 @@
+      throws SynchronisedInstallerException {
+    ServiceTracker st = getServiceTracker(bundleContext, getServiceIdFilter(servicePid));
+    st.open(true);
+
+    Callable<Boolean> isServiceAvailable =
+        () -> {
+          LOGGER.debug("Waiting for service with pid [{}] to be registered", servicePid);
+          return st.getService() != null;
+        };
+
+    try {
+      wait(
+          isServiceAvailable,
+          maxWaitTime,
+          DEFAULT_POLLING_INTERVAL,
+          "Managed service failed to appear after [" + maxWaitTime + "] ms.");
+    } finally {
+      st.close();
+    }
+  }
+
+  @Override
+  public void installFeatures(String feature, String... additionalFeatures)
+      throws SynchronisedInstallerException {
+    installFeatures(DEFAULT_MAX_FEATURE_WAIT, feature, additionalFeatures);
+  }
+
+  @Override
+  public void installFeatures(long maxWaitTime, String feature, String... additionalFeatures)
+      throws SynchronisedInstallerException {
+
+    Set<String> featuresToInstall =
+        getFeaturesFromNames(feature, additionalFeatures)
+            .filter(f -> !featuresService.isInstalled(f))
+            .map(Feature::getName)
+            .collect(Collectors.toSet());
+
+    LOGGER.info("Installing the following features: [{}]", String.join(", ", featuresToInstall));
+    long startTime = System.currentTimeMillis();
+    try {
+      AccessController.doPrivileged(
+          (PrivilegedExceptionAction<Void>)
+              () -> {
+                featuresService.installFeatures(
+                    featuresToInstall, EnumSet.of(FeaturesService.Option.NoAutoRefreshBundles));
+                return null;
+              });
+    } catch (Exception e) {
+      throw new SynchronisedInstallerException(
+          "Failed to install features [" + String.join(",", featuresToInstall) + "]", e);
+    }
+
+    waitForBundles(maxWaitTime);
+    LOGGER.info("Finished installing features in {} ms", (System.currentTimeMillis() - startTime));
+  }
+
+  @Override
+  public void uninstallFeatures(String feature, String... additionalFeatures)
+      throws SynchronisedInstallerException {
+    uninstallFeatures(DEFAULT_MAX_FEATURE_WAIT, feature, additionalFeatures);
+  }
+
+  @Override
+  public void uninstallFeatures(long maxWaitTime, String feature, String... additionalFeatures)
+      throws SynchronisedInstallerException {
+    Set<String> featuresToUninstall =
+        getFeaturesFromNames(feature, additionalFeatures)
+            .filter(f -> featuresService.isInstalled(f))
+            .map(Feature::getName)
+            .collect(Collectors.toSet());
+
+    LOGGER.info(
+        "Uninstalling the following features: [{}]", String.join(", ", featuresToUninstall));
+    long startTime = System.currentTimeMillis();
+    try {
+      AccessController.doPrivileged(
+          (PrivilegedExceptionAction<Void>)
+              () -> {
+                featuresService.uninstallFeatures(
```

Comment:
I think we have a chance here to address one of the issue in Karaf about feature installation and required as we did in the profile migratable. Here is the comment from that code
```
    // to work around a bug in Karaf where for whatever reasons, it checks if a feature is
    // required to determine if it is installed and therefore can be uninstalled, we will first
    // go through and mark them all required
    //
    // see Karaf class: org.apache.karaf.features.internal.service.FeaturesServiceImpl.java
    // in the uninstallFeature() method where it checks the requirements table and if not found
    // in there, it later reports it is not installed even though it is actually installed
    //
    // Update: turns out that what Karaf means by install is "require this feature" and by uninstall
    // is "no longer require this feature". That being said, the uninstallFeature() does a bit more
    // than simply marking the feature no longer required since it actually also removes all traces
    // of the feature internally. So to properly remove a feature, it has to first be marked
    // required. which can technically be done either by updating the requirements for the feature
    // (as done below) or actually calling installFeature() again. We opted for the former since we
    // already deal with updating requirements to match the 'required' state of the feature from
    // the original system
```
And to address this shortcoming, we simply do this update the requirements before uninstalling. Check `org.codice.ddf.admin.application.service.migratable.FeatureProcessor`